counsel.el

```diff
@@ -2278,6 +2278,14 @@ PREFIX is used to create the key."
                                            (cdr elm))))))))
              alist))
 
+(defun counsel-imenu-categorize-functions (items)
+  "Categorize all the functions of imenu."
+  (let* ((others (remove-if-not (lambda (x) (listp (cdr x))) items))
+         (functions (remove-if (lambda (x) (listp (cdr x))) items)))
+    (if functions
+        (append others `(("Function" ,@functions)))
+      items)))
```

Comment:
The Swiper project uses `cl-lib`, not `cl`, so you should use the `cl-` namespace. You can also simplify this code a bit:

```el
(defun counsel-imenu-categorize-functions (items)
  "Categorize all the functions of imenu."
  (let ((fns (cl-remove-if #'listp items :key #'cdr))
        (others (cl-remove-if #'nlistp items :key #'cdr)))
    (if fns
        (nconc others `(("Function" ,@fns)))
      items)))
```

or even

```el

(defun counsel-imenu-categorize-functions (items)
  "Categorize all the functions of imenu."
  (let ((fns (cl-remove-if #'listp items :key #'cdr)))
    (if fns
        (nconc (cl-remove-if #'nlistp items :key #'cdr)
               `(("Function" ,@fns)))
      items)))
```